manager/manager.go

```diff
@@ -346,19 +363,34 @@ func (m *Manager) Run(parent context.Context) error {
 	// Set the raft server as serving for the health server
 	healthServer.SetServingStatus("Raft", api.HealthCheckResponse_SERVING)
 
-	if err := m.raftNode.JoinAndStart(ctx); err != nil {
+	if err := m.joinAndStart(ctx); err != nil {
 		return errors.Wrap(err, "can't initialize raft node")
 	}
+	m.keyRotator.SetKeyWriter(m.config.SecurityConfig.KeyWriter())
 
 	localHealthServer.SetServingStatus("ControlAPI", api.HealthCheckResponse_SERVING)
 
 	close(m.started)
 
+	go m.watchForKEKChanges(ctx)
+
 	go func() {
+		if m.keyRotator.GetNewKey() != nil {
+			m.keyRotator.rotationChan <- struct{}{}
+		}
 		err := m.raftNode.Run(ctx)
 		if err != nil {
 			log.G(ctx).Error(err)
 			m.Stop(ctx)
+			if err == raft.ErrMemberRemoved {
+				if err := m.config.SecurityConfig.KeyWriter().UpdateHeaders(func(headers map[string]string, _ []byte) error {
+					delete(headers, defaultRaftDEKKey)
+					delete(headers, defaultRaftDekKeyPending)
+					return nil
+				}); err != nil {
+					log.G(ctx).Error(errors.Wrap(err, "unable to remove raft DEK from TLS certificate"))
+				}
+			}
```

Comment:
Currently updating the TLS certificates preserves all the headers, because we need the DEKs preserved when the manager TLS certs. I can add that to the writer, to check the cert and if the role has changed to wipe out all the headers.